### PR TITLE
Clean up remaining tests under JIT/Methodical using Environment.Exit

### PR DESCRIPTION
**Pull request by trylek**
I have manually rewritten the few remaining tests under JIT/Methodical to stop using Environment.Exit
to report their results (this was preventing in-process execution and subsequently their merging).

Thanks

Tomas

/cc @dotnet/jit-contrib

### CODE REVIEW

File: src/tests/JIT/Methodical/tailcall/test_implicit.il [hunk 1]
```diff
@@ -80,6 +84,7 @@ ret
     IL_0049:  br.s       IL_004b
 
     IL_004b:  ldloc.0
+exit:
```
Comment (BruceForstall) [inline src/tests/JIT/Methodical/tailcall/test_implicit.il:87]: Is this new label unreferenced?

**Review by BruceForstall — APPROVED** [review]
test_implicit is tricky, but AFAICT the change looks ok.

File: src/tests/JIT/Methodical/tailcall/test_implicit.il (hunk 1 above)
Comment (trylek) [reply src/tests/JIT/Methodical/tailcall/test_implicit.il:87]: Hmm, you're right, I have no idea why I put it there. Thanks for spotting. will remove.

Comment (trylek) [reply src/tests/JIT/Methodical/tailcall/test_implicit.il]: Fixed in 2nd commit.

**Pull request merged**